src/cli/SwaCliTaskProvider.ts

```diff
@@ -75,49 +76,51 @@ export class SwaTaskProvider implements TaskProvider {
     }
 
     private createSwaConfigTask(workspaceFolder: WorkspaceFolder, configurationName?: string): Task {
-        const command = configurationName ? `start ${configurationName}` : 'start';
-        const task = new Task({
-            type: shell,
-            command,
-        }, workspaceFolder, command, swa, new ShellExecution(`swa ${command}`), swaWatchProblemMatcher);
-        task.isBackground = true;
-
-        return task;
+        const args: string[] = ['start'];
+        if (configurationName) {
+            args.push(configurationName);
+        }
+
+        return new SwaTask(
+            workspaceFolder,
+            args.join(' '),
+            new SwaShellExecution(args)
+        );
     }
 
     private createSwaCliTask(workspaceFolder: WorkspaceFolder, label: string, options: Pick<SWACLIOptions, 'context' | 'apiLocation' | 'run' | 'appLocation'>): Task {
 
-        const addArg = <T>(object: T, property: keyof T, name?: string, quote?: boolean): string => {
-            return object[property] ? quote ? `--${name ?? property.toString()}=\"${object[property]}\"` : `--${name ?? property.toString()}=${object[property]}` : '';
+        const addArg = <T extends Record<string, string>>(object: T, property: keyof T, name?: string): string[] => {
+            const args: string[] = [];
+            if (object[property]) {
+                args.push(`--${name ?? property.toString()}`);
+                args.push(object[property] as string);
+            }
+            return args;
         };
 
-        const args: string[] = [addArg(options, 'appLocation', 'app-location'), addArg(options, 'apiLocation', 'api-location'), addArg(options, 'run', 'run', true)];
-
-        // Increase devserver timeout to 3x default. See https://github.com/microsoft/vscode-azurestaticwebapps/issues/574#issuecomment-965590774
-        args.push('--devserver-timeout=90000');
-        const command = `swa start ${options.context} ${args.join(' ')}`;
-        const task = new Task(
-            {
-                type: shell,
-                command,
-            },
+        const args: string[] = [
```

Comment:
You may want to use VSCode's `ShellQuotedString` instead of `string`. You can be more specific about _how_ the argument should be quoted when it is executed in `ShellExecution`. It's not mandatory, just something to look at.